pkg/chunked/storage_linux.go

```diff
@@ -146,40 +146,48 @@ func (c *chunkedDiffer) convertTarToZstdChunked(destDirectory string, payload *o
 func GetDiffer(ctx context.Context, store storage.Store, blobDigest digest.Digest, blobSize int64, annotations map[string]string, iss ImageSourceSeekable) (graphdriver.Differ, error) {
 	pullOptions := store.PullOptions()
 
+	convertImages := parseBooleanPullOption(pullOptions, "convert_images", false)
+
 	if !parseBooleanPullOption(pullOptions, "enable_partial_images", true) {
-		return nil, errors.New("enable_partial_images not configured")
+		return nil, internal.ErrPartialContentNotAvailable
 	}
 
 	zstdChunkedTOCDigestString, hasZstdChunkedTOC := annotations[internal.ManifestChecksumKey]
 	estargzTOCDigestString, hasEstargzTOC := annotations[estargz.TOCJSONDigestAnnotation]
 
 	if hasZstdChunkedTOC && hasEstargzTOC {
-		return nil, errors.New("both zstd:chunked and eStargz TOC found")
+		if !convertImages {
+			return nil, fmt.Errorf("both zstd:chunked and eStargz TOC found: %w", internal.ErrPartialContentNotAvailable)
+		}
+		// If a conversion is required, ignore both TOCs and retrieve the entire layer.
+		hasZstdChunkedTOC, hasEstargzTOC = false, false
 	}
 
 	if hasZstdChunkedTOC {
 		zstdChunkedTOCDigest, err := digest.Parse(zstdChunkedTOCDigestString)
-		if err != nil {
-			return nil, fmt.Errorf("parsing zstd:chunked TOC digest %q: %w", zstdChunkedTOCDigestString, err)
+		if err == nil {
+			return makeZstdChunkedDiffer(store, blobSize, zstdChunkedTOCDigest, annotations, iss, pullOptions)
+		}
+		if !convertImages {
+			return nil, fmt.Errorf("parsing zstd:chunked TOC digest %q: %w", zstdChunkedTOCDigestString, internal.ErrPartialContentNotAvailable)
```

Comment:
These error messages don’t look all that useful — and lose the actual text describing the reason for the failure.

WDYT about
```go
type Err$name struct {
   underlying error
}
```
implementing `Unwrap`? Then we can “mark” any error without affecting the type… and we can have a separate design conversation on whether the wrapping type should add any error text at all.

---

Alternatively, this works:
```go
return fmt.Errorf("%.0w parsing zstd:chunked TOC digest %q: %w", internal.ErrPartialContentNotAvailable, …)
```
where `%.0w` adds the error to `Unwrap` without affecting the text. (OTOH, as a matter of aesthetics / general downsides, I generally dislike the multi-error `Unwrap` — and we would need a comment for `%.0w`, explicit wrapping is easier to understand).